### PR TITLE
Add a configurable AndroidLogging API

### PR DESCRIPTION
**Pull request by yschimke**
Breaking up the larger okhttp-android PR.

Allow android clients to use Android Log instead of Java logging.

### CODE REVIEW

File: okhttp-android/src/main/kotlin/okhttp3/android/AndroidLogging.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package okhttp3.android
+
+import android.util.Log
+import okhttp3.logging.HttpLoggingInterceptor
+import okhttp3.logging.LoggingEventListener
+
+fun LoggingEventListener.Companion.AndroidLogging(
```
Comment (swankjesse) [inline okhttp-android/src/main/kotlin/okhttp3/android/AndroidLogging.kt:7]: Why an extension function? What does it look like for Java language callers?

Comment (swankjesse) [reply okhttp-android/src/main/kotlin/okhttp3/android/AndroidLogging.kt:7]: I’m also partial to Timber, but I don’t think it’s necessarily worth the trouble to offer a Timber integration.

Comment (yschimke) [reply okhttp-android/src/main/kotlin/okhttp3/android/AndroidLogging.kt:7]: I kind of figure Kotlin is a safe choice on Android.  Can refactor. The point of the PR is to support logging with the feature set of Android. I don't mind the API.

The extension function makes it appear to be part of the logging library, not a secondary thing to discover.

File: okhttp-android/src/main/kotlin/okhttp3/android/AndroidLogging.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package okhttp3.android
```
Comment (swankjesse) [inline okhttp-android/src/main/kotlin/okhttp3/android/AndroidLogging.kt:1]: supernit: copyright header

**Review by swankjesse — APPROVED** [review]

**Comment by yschimke** [comment]
@swankjesse added some tests and minimal docs.

**Pull request merged**